apps/hannk/Makefile

```diff
@@ -6,26 +6,64 @@ include ../support/Makefile.inc
 # builds but easier to enable it for everything.
 CXXFLAGS += -Wno-unused-private-field -fno-exceptions -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wunused-variable -I$(MAKEFILE_DIR)
 
+BENCHMARK_OUT = benchmark
+ifeq (hexagon-32-qurt,$(findstring hexagon-32-qurt,$(HL_TARGET)))
+	# Building benchmark application as shared object instead of elf for
+	# hexagon target.
+	BENCHMARK_OUT := $(BENCHMARK_OUT).so
+	BENCHMARK_HEXAGON_FLAGS = -shared -fPIC -G0
+	HEXAGON_STUBS = $(BIN)/$(HL_TARGET)/stubs.o
+	TEST ?= test/misc/MUL.tflite
+	TEST_NAME = $(notdir $(TEST))
+endif
+
+
 ifneq (,$(findstring -O,$(OPTIMIZE)))
 	CXXFLAGS += -DNDEBUG
 endif
 
 MAKEFILE_DIR=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
 
-.PHONY: all build clean test delegate
+.PHONY: all build clean test delegate test-hexagon-sim test-hexagon-device
 all: build
 
 # .SECONDARY with no prerequisites causes all targets to be treated as secondary
 # (i.e., no target is removed because it is considered intermediate).
 .SECONDARY:
 
 build: \
-	$(BIN)/$(HL_TARGET)/benchmark \
+	$(BIN)/$(HL_TARGET)/$(BENCHMARK_OUT) \
 	$(BIN)/$(HL_TARGET)/compare_vs_tflite
 
 test: compare_vs_tflite
 	$(BIN)/$(HL_TARGET)/compare_vs_tflite test/*/* --benchmark 0
 
+test-hexagon-sim: $(BIN)/$(HL_TARGET)/$(BENCHMARK_OUT)
+	@mkdir -p $@
+	echo "Benchmarking $(TEST)..."
+	echo "$(HEXAGON_SDK_ROOT)/rtos/qurt/computev65/debugger/lnx64/qurt_model.so" > $(BIN)/$(HL_TARGET)/osam.cfg
+	echo "$(DEFAULT_HEXAGON_TOOLS_ROOT)/Tools/lib/iss/qtimer.so --csr_base=0xFC900000 --irq_p=1 --freq=19200000 --cnttid=1" > $(BIN)/$(HL_TARGET)/q6ss.cfg
```

Comment:
How stable are these specific files/constants? Are they going to break often?

I have this comment on many things in this PR... It seems like a lot of the config stuff happening in this PR should be built into the tools themselves, so we don't need to bake so many things that could potentially break into our build?

Reply:
The constants are stable for the same target (same qurt ‘variant’, like ‘computev65’). The configs are not present in the tools. The sdk examples also do the same thing to run.